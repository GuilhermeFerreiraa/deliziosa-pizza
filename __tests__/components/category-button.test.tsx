import CategoryButton from "@/components/category-button";
import React from "react";
import renderer from "react-test-renderer";

describe("CategoryButton Component", () => {
 it("renders with title", () => {
  const component = renderer.create(
   <CategoryButton title="Test Category" />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
 });

 it("renders as selected", () => {
  const component = renderer.create(
   <CategoryButton title="Selected Category" isSelected />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
 });
});
