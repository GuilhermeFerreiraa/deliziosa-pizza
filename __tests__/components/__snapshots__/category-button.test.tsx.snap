// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryButton Component renders as selected 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#BF0A25",
      },
      {
        "paddingLeft": 16,
        "paddingRight": 16,
      },
      {
        "justifyContent": "center",
      },
      {
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
      },
      {
        "height": 40,
      },
      {
        "borderBottomWidth": 2,
        "borderLeftWidth": 2,
        "borderRightWidth": 2,
        "borderTopWidth": 2,
      },
      {
        "borderBottomColor": "#BF0A25",
        "borderLeftColor": "#BF0A25",
        "borderRightColor": "#BF0A25",
        "borderTopColor": "#BF0A25",
      },
      {
        "backgroundColor": "#f1f5f9",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#f1f5f9",
        },
        {
          "fontFamily": "Inter_500Medium",
        },
        {
          "fontSize": 14,
          "lineHeight": 20,
        },
        {
          "color": "#BF0A25",
        },
        {
          "fontFamily": "Inter_600SemiBold",
        },
      ]
    }
  >
    Selected Category
  </Text>
</View>
`;

exports[`CategoryButton Component renders with title 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#BF0A25",
      },
      {
        "paddingLeft": 16,
        "paddingRight": 16,
      },
      {
        "justifyContent": "center",
      },
      {
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
      },
      {
        "height": 40,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#f1f5f9",
        },
        {
          "fontFamily": "Inter_500Medium",
        },
        {
          "fontSize": 14,
          "lineHeight": 20,
        },
      ]
    }
  >
    Test Category
  </Text>
</View>
`;
